core/server/inventory.go

```diff
@@ -446,3 +436,26 @@ func parseInventoryFromUnstructured(obj *unstructured.Unstructured) ([]*unstruct
 
 	return objects, nil
 }
+
+func secretNameFromHelmRelease(helmRelease *helmv2.HelmRelease) *client.ObjectKey {
+	if helmRelease.Status.History == nil && helmRelease.Status.LastReleaseRevision == 0 {
+		return nil
+	}
+
+	releaseRevision := helmRelease.Status.LastReleaseRevision
+
+	if helmRelease.Status.History != nil {
```

Comment:
This check is redundant, `Latest()` itself will check if `len(in) == 0` which will be `true` for `nil`.